Mathlib/CategoryTheory/Bicategory/Grothendieck.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2024 Calle Sönne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Calle Sönne
+-/
+
+import Mathlib.CategoryTheory.Bicategory.Functor.Pseudofunctor
+import Mathlib.CategoryTheory.Bicategory.LocallyDiscrete
+import Mathlib.CategoryTheory.Category.Cat
+
+/-!
+# The Grothendieck construction
+
+Given a category `𝒮` and any pseudofunctor `F` from `𝒮ᵒᵖ` to `Cat`, we associate to it a category
+`∫ F`, equipped with a functor `∫ F ⥤ 𝒮`.
+
+The category `∫ F` is defined as follows:
+* Objects: pairs `(S, a)` where `S` is an object of the base category and `a` is an object of the
+  category `F(S)`.
+* Morphisms: morphisms `(R, b) ⟶ (S, a)` are defined as pairs `(f, h)` where `f : R ⟶ S` is a
+  morphism in `𝒮` and `h : b ⟶ F(f)(a)`
+
+The projection functor `∫ F ⥤ 𝒮` is then given by projecting to the first factors, i.e.
+* On objects, it sends `(S, a)` to `S`
+* On morphisms, it sends `(f, h)` to `f`
+
+## TODO
+1. Implement functoriality for the Grothendieck construction.
+2. Obtain the results in `CategoryTheory.Grothendieck` as a specialization of these results.
+
+## References
+[Vistoli2008] "Notes on Grothendieck Topologies, Fibered Categories and Descent Theory" by
+Angelo Vistoli
+
+-/
+
+namespace CategoryTheory
+
+universe w v₁ v₂ v₃ u₁ u₂ u₃
+
+open CategoryTheory Functor Category Opposite Discrete Bicategory
+
+variable {𝒮 : Type u₁} [Category.{v₁} 𝒮] {F : Pseudofunctor (LocallyDiscrete 𝒮ᵒᵖ) Cat.{v₂, u₂}}
+
+/-- The type of objects in the fibered category associated to a presheaf valued in types. -/
+structure Pseudofunctor.Grothendieck (F : Pseudofunctor (LocallyDiscrete 𝒮ᵒᵖ) Cat.{v₂, u₂}) where
+  /-- The underlying object in the base category. -/
+  base : 𝒮
+  /-- The object in the fiber of the base object. -/
+  fiber : F.obj ⟨op base⟩
+
+/-- Notation for the Grothendieck category associated to a pseudofunctor `F`. -/
+prefix:75 "∫ " => Pseudofunctor.Grothendieck
+
+namespace Pseudofunctor.Grothendieck
```

Comment:
The notation "∫" is used in other parts of mathlib. The following should make "open Pseudofunctor.Grothendieck" necessary to get this notation:
```suggestion
namespace Pseudofunctor.Grothendieck

scoped prefix:75 "∫ " => Pseudofunctor.Grothendieck
```

Reply:
Thanks!! I'm somewhat new to defining notation :)